flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java

```diff
@@ -760,6 +767,79 @@ public void invoke() throws Exception {
 		}
 	}
 
+	@Test
+	public void testTaskInterruptionAndTerminationOnShutdown() throws Exception {
+		final CompletableFuture<Void> taskSlotTableStarted = new CompletableFuture<>();
+		final TaskSlotTable taskSlotTable = new TaskSlotTableWithStartFuture(taskSlotTableStarted, timerService);
+		final JobMasterId jobMasterId = JobMasterId.generate();
+		final AllocationID allocationId = new AllocationID();
+		final TaskDeploymentDescriptor taskDeploymentDescriptor =
+			createTaskDeploymentDescriptor(TestInterruptableInvokable.class, allocationId);
+		final TaskExecutor taskExecutor = createTaskExecutor(new TaskManagerServicesBuilder()
+			.setTaskSlotTable(taskSlotTable)
+			.setJobManagerTable(createJobManagerTableWithOneJob(jobMasterId))
+			.build());
+
+		try {
+			taskExecutor.start();
+			taskSlotTableStarted.get();
+			taskSlotTable.allocateSlot(0, jobId, allocationId, timeout);
```

Comment:
This is dangerous, as we are mutating state from the testing thread which is actually owned by the `TaskExecutor's` rpc main thread. Better to call `TaskExecutorGateway#requestSlot` after registering a `ResourceManagerGateway`.